### PR TITLE
PYTHON-4703 MongoClient should default to connect=False on FaaS environments

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -42,13 +42,19 @@ PyMongo 4.9 brings a number of improvements including:
 - Fixed a bug where PyMongo would raise ``InvalidBSON: date value out of range``
   when using :attr:`~bson.codec_options.DatetimeConversion.DATETIME_CLAMP` or
   :attr:`~bson.codec_options.DatetimeConversion.DATETIME_AUTO` with a non-UTC timezone.
+- The default value for ``connect`` in ``MongoClient`` is changed to ``False`` when running on
+  Function-as-a-service (FaaS).  On some FaaS systems, there is a ``fork()`` operation at function
```
Comment (ShaneHarvey) [inline doc/changelog.rst:46]: Suggest adding `Function-as-a-service (FaaS) like AWS Lambda, Google Cloud Functions, and Microsoft Azure Functions.`

File: test/lambda/mongodb/app.py [hunk 2]
```diff
@@ -18,6 +19,9 @@
     ServerHeartbeatListener,
 )
 
+# Ensure there are no warnings raised in normal operation.
+warnings.simplefilter("error")
```
Comment (ShaneHarvey) [inline test/lambda/mongodb/app.py:23]: Nice!

File: pymongo/asynchronous/mongo_client.py [hunk 3]
```diff
@@ -803,7 +803,10 @@ def __init__(
         if tz_aware is None:
             tz_aware = opts.get("tz_aware", False)
         if connect is None:
-            connect = opts.get("connect", True)
+            # Default to connect=True unless on a FaaS system, which might use fork.
```
Comment (ShaneHarvey) [inline pymongo/asynchronous/mongo_client.py:806]: Suggest adding a versionchanged and updating the docstring for `connect`.

File: doc/changelog.rst (hunk 1 above)
Comment (blink1073) [reply doc/changelog.rst:46]: Done

File: pymongo/asynchronous/mongo_client.py (hunk 3 above)
Comment (blink1073) [reply pymongo/asynchronous/mongo_client.py:810]: Done

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**